cli/flox/src/commands/init/python.rs

```diff
@@ -438,6 +437,18 @@ impl Provider for PoetryPyProject {
                 source "$(poetry env info --path)/bin/activate""#}
                 .to_string(),
             ),
+            profile_fish: Some(
+                indoc! {r#"
+                echo "Activating poetry virtual environment" >&2
+                source "$(poetry env info --path)/bin/activate.fish""#}
+                .to_string(),
+            ),
+            profile_tcsh: Some(
+                indoc! {r#"
+                echo "Activating poetry virtual environment" >&2
+                source "$(poetry env info --path)/bin/activate.csh""#}
+                .to_string(),
```

Comment:
question blocking: how prominent do we want tcsh to be in docs and init hooks? I'm not really sure how much it's used, so I'm not sure if most people will see it as boilerplate or actually helpful

Reply:
agree, just fish is good

Reply:
Thinking about this further I disagree. Flox has a real application in companies seeking to wrap their arms around legacy technologies, and seeing as we are autogenerating template for multiple shells I don't see why we wouldn't do that for all supported shells. It seems snobbish, particularly as direnv and Python venvs both support tcsh, and at a minimum removing this would break our Python environment tests. With that I'd like to retain this logic. 

Can we resolve the conversation on that basis? Thanks.

Reply:
I think we can support tcsh without putting it front and center in our templates. I agree that there should be a line for tcsh in the "shells we support table" just like Python venvs https://docs.python.org/3/library/venv.html. But I don't think we need to give tcsh a few lines in all of our docs/templates. Unless it's more widely used than I realize

Reply:
> I think we can support tcsh without putting it front and center in our templates

If the code generation doesn't occur for a given shell then you cannot say that automatic python venvs are supported for that shell. In other words if you remove this bit of template generation you'll break this feature for tcsh, in which case you cannot say you support tcsh. Or am I missing something?

Reply:
To be clear, when we generate the default template there is no mention of tcsh, or bash, or fish, or zsh - the only mention of those shells is in the documentation. Is that what you're wanting to remove, or are you wanting to remove this automatic python manifest support?

Reply:
I meant both - I'm not sure we need the tcsh support in language specific flox init, and I'm not sure we need the tcsh block in `manifest.toml.md` (although we should definitely say we have a `profile.tcsh` section).

We don't really document our init hooks at all, they're just recommendations. And I'm guessing it's an okay recommendation for most people to not add a tcsh hook.

Reply:
If you look ahead we may eventually support other shells like `dash`, `ksh`, etc. In short we may support an arbitrary number of shells in the future, and I think it's reasonable to draw some line (probably based on popularity) as to which shells are the defaults for automatic setup, documentation, etc so as not to clutter up the documentation, templates, etc.

I _also_ think it wouldn't be too hard to add a `--shells` flag to `init` to be specified along with `--auto-setup`.

Reply:
>I'm not sure we need the tcsh support in language specific flox init, and I'm not sure we need the tcsh block in manifest.toml.md (although we should definitely say we have a profile.tcsh section).

>If you look ahead we may eventually support other shells like dash, ksh, etc. In short we may support an arbitrary number of shells in the future, and I think it's reasonable to draw some line (probably based on popularity) as to which shells are the defaults for automatic setup, documentation, etc so as not to clutter up the documentation, templates, etc.

from a product standpoint, I agree with this. it's about prominence. bash, zsh, fish should be more prominent than the others IMO.

Reply:
* removed example tcsh stanza from `manifest.toml(5)` man page with commit a89a7165a25ff2e401a2f056b255a9ddaee7ade1
* when performing a `flox init` there is no sample `[profile.tcsh]` stanza created, and nor is there for any of the other [3] shells (this was a point of confusion as some thought there would be)
* when performing a `flox init --auto-setup` for a python project specifically, manifest stanzas are still being autogenerated for each of the [4] supported shells, and `tests/python.bats` has been updated to test all 4 of these
    - in essence this is what it means for a shell to be fully supported, and until we develop a facility for only performing initialization for selected shells as Zach proposed [above](https://github.com/flox/flox/pull/1347/files#r1622699412) this is in my view the most correct behaviour

Reply:
LGTM